extensions/amazon-lambda/common-runtime/src/main/java/io/quarkus/amazon/lambda/runtime/AbstractLambdaPollLoop.java

```diff
@@ -57,7 +63,24 @@ public void run() {
                         }
                         try {
                             String requestId = requestConnection.getHeaderField(AmazonLambdaApi.LAMBDA_RUNTIME_AWS_REQUEST_ID);
+                            if (requestConnection.getResponseCode() != 200) {
+                                // connection should be closed by finally clause
+                                continue;
+                            }
                             try {
+                                if (LambdaHotReplacementRecorder.enabled) {
+                                    hotRestart.set(true);
+                                    if (LambdaHotReplacementRecorder.checkHotReplacement()) {
+                                        // hot replacement happened in dev mode
+                                        // so we requeue the request as quarkus will restart
+                                        // and the message will not be processed
+                                        // FYI: this requeue endpoint is something only the mock event server implements
+                                        requeue(requestId);
+                                        continue;
+                                    } else {
+                                        hotRestart.set(false);
```

Comment:
Shouldt this be in a try/finally rather than an else block? As is it remains false until the next request that does not do a hot replacement?

Reply:
No, it should not be a try/finally.  this flag exists to prevent the shutdown hook from interrupting the polling thread.  If the polling thread is interrupted, then quarkus restart will fail as the polling thread is doing the restart on hot replacement